e2e/internal/keyexec/keyexec.go

```diff
@@ -0,0 +1,361 @@
+
+	exportCmd := []string{"key", "import", kpath}
+
+	cmd := exec.Command(testenv.CmdPath, exportCmd...)
+	cmd.Stdin = c.Tty()
+
+	buf := bytes.NewBuffer(nil)
+	cmd.Stderr = buf
+	cmd.Stdout = buf
+
+	go func() {
+		c.ExpectEOF()
+	}()
+
+	err = cmd.Start()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Send the passcode to singularity
+	c.Send("e2etests\n")
+	c.Send("e2etests\n")
+	c.Send("e2etests\n")
+
+	err = cmd.Wait()
+	cm := fmt.Sprintf("%s %s", testenv.CmdPath, strings.Join(exportCmd, " "))
+
+	return cm, buf.Bytes(), err
+}
+
+// ExportPrivateKey will import a private key from kpath.
+func ExportPrivateKey(t *testing.T, kpath, num string, armor bool) (string, []byte, error) {
+	e2e.LoadEnv(t, &testenv)
+
+	c, err := expect.NewConsole(expect.WithStdin(os.Stdin))
+	if err != nil {
+		panic(err)
+	}
+	defer c.Close()
+
+	exportCmd := []string{"key", "export", "--secret"}
+
+	if armor {
+		exportCmd = append(exportCmd, "--armor")
+	}
+
+	exportCmd = append(exportCmd, kpath)
+
+	outErr := bytes.NewBuffer(nil)
+
+	cmd := exec.Command(testenv.CmdPath, exportCmd...)
+	cmd.Stdin = c.Tty()
+
+	cmd.Stderr = outErr
+	cmd.Stdout = outErr
+
+	go func() {
+		c.ExpectEOF()
+	}()
+
+	err = cmd.Start()
+	if err != nil {
+		t.Fatalf("unable to run command: %v", err)
+	}
+
+	c.Send(num)
+	c.Send("e2etests\n")
+
+	err = cmd.Wait()
+	cm := fmt.Sprintf("%s %s", testenv.CmdPath, strings.Join(exportCmd, " "))
+
+	return cm, outErr.Bytes(), err
+}
+
+// CorruptKey will take a ASCII key (kpath) and change some chars in it (corrupt it).
+func CorruptKey(t *testing.T, kpath string) {
+	input, err := ioutil.ReadFile(kpath)
+	if err != nil {
+		t.Fatalf("Unable to read file: %v", err)
+	}
+
+	lines := strings.Split(string(input), "\n")
+
+	for i, line := range lines {
+		if strings.Contains(line, "B") {
+			lines[i] = "P"
+		}
+	}
+	output := strings.Join(lines, "\n")
+	err = ioutil.WriteFile(kpath, []byte(output), 0644)
+	if err != nil {
+		t.Fatalf("Unable to write to file: %v", err)
+	}
+}
+
+// RunKeyCmd will run a 'singularty key' command, with any args that are set in commands.
+func RunKeyCmd(t *testing.T, cmdPath string, commands []string, file, stdin string) (string, []byte, error) {
+	argv := []string{"key"}
+	argv = append(argv, commands...)
+
+	if file != "" {
+		argv = append(argv, file)
+	}
+
+	cmd := fmt.Sprintf("%s %s", cmdPath, strings.Join(argv, " "))
+	execKey := exec.Command(cmdPath, argv...)
+
+	stdinRun, err := ioutil.TempFile("", "")
+	if err != nil {
+		t.Fatalf("%v", err)
+	}
+	defer stdinRun.Close()
+
+	_, err = io.WriteString(stdinRun, stdin)
+	if err != nil {
+		t.Fatalf("%v", err)
+	}
+
+	_, err = stdinRun.Seek(0, os.SEEK_SET)
+	if err != nil {
+		t.Fatalf("%v", err)
+	}
+
+	execKey.Stdin = stdinRun
+	out, err := execKey.CombinedOutput()
+
+	return cmd, out, err
+}
+
+// QuickTestKey will export a private, and public key (0), and then import them. This is used after
+// generating a newpair.
+func QuickTestKey(t *testing.T) {
```

Comment:
Better to call it `QuickTestExportImportKey` to know exactly what this test is doing beforehand